### PR TITLE
KOGITO-405: [DMN Designer] Unit tests: kie-wb-common-dmn-webapp-kogito-common

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/KOGITO-405

Just Unit Tests and (in a few places) a little refactor to make testing easier.

### CODE REVIEW

**Comment by manstis** [comment]
The [class](https://sonarcloud.io/component_measures?id=org.kie.workbench%3Akie-wb-common&metric=new_branch_coverage&pullRequest=3009&selected=org.kie.workbench%3Akie-wb-common%3Akie-wb-common-dmn%2Fkie-wb-common-dmn-webapp-kogito-common%2Fsrc%2Fmain%2Fjava%2Forg%2Fkie%2Fworkbench%2Fcommon%2Fdmn%2Fwebapp%2Fkogito%2Fcommon%2Fclient%2Feditor%2FAbstractDMNDiagramEditor.java&view=list) Sonar is complaining about having zero coverage is a base abstract class. Tests for the sub-classes of the base abstract class provide 88% coverage.

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Comment by manstis** [comment]
@jomarko Let's get these tests in as soon as possible... I'm more than conscious that test coverage following merge of DROOLS-3727 is incomplete and this type of PR only helps correct the imbalance.

**Comment by manstis** [comment]
@jomarko Gentle reminder.. please don't forget this PR.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/DMNEditorMenuSessionItemsTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    public void testDisableMenu() {
+        testMenu(false);
+    }
+
+    private void testMenu(final boolean enabled) {
+        doNothing().when(sessionItems).superSetEnabled(enabled);
+
+        sessionItems.setEnabled(enabled);
+
+        verify(sessionItems).setItemEnabled(PerformAutomaticLayoutCommand.class, enabled);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/DMNEditorMenuSessionItemsTest.java]: and what about
```java
setItemEnabled(ClearSessionCommand.class, enabled);
        setItemEnabled(VisitGraphSessionCommand.class, enabled);
        setItemEnabled(SwitchGridSessionCommand.class, enabled);
        setItemEnabled(ValidateSessionCommand.class, enabled);
        setItemEnabled(ExportToJpgSessionCommand.class, enabled);
        setItemEnabled(ExportToPngSessionCommand.class, enabled);
        setItemEnabled(ExportToSvgSessionCommand.class, enabled);
        setItemEnabled(ExportToPdfSessionCommand.class, enabled);
        setItemEnabled(ExportToRawFormatSessionCommand.class, enabled);
        setItemEnabled(DeleteSelectionSessionCommand.class, false);
        setItemEnabled(UndoSessionCommand.class, false);
        setItemEnabled(RedoSessionCommand.class, false);
        setItemEnabled(CopySelectionSessionCommand.class, false);
        setItemEnabled(CutSelectionSessionCommand.class, false);
        setItemEnabled(PasteSelectionSessionCommand.class, false);
```

do we assert them somewhere?

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/workarounds/IsKogitoTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (gitgabrio) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/workarounds/IsKogitoTest.java]: @manstis 
TBH I think that "IsKogito" has to be completely removed in the near future (and so this test)

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/AbstractDMNDiagramEditorTest.java [hunk 3]
```diff
@@ -222,11 +232,23 @@
     @Mock
     protected Path root;
 
+    @Mock
+    protected Path path;
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/AbstractDMNDiagramEditorTest.java]: Removed.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/DMNEditorMenuSessionItemsTest.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/DMNEditorMenuSessionItemsTest.java]: Added. I also added in a similar existing test.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**